control-plane/api-gateway/consul/config_entry_translation.go

```diff
@@ -0,0 +1,100 @@
+package consul
+
+import (
+	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/hashicorp/consul/api"
+	consulAPI "github.com/hashicorp/consul/api"
+)
+
+const (
+	metaKeyManagedBy       = "managed-by"
+	metaValueManagedBy     = "consul-k8s-gateway-controller"
+	metaKeyKubeNS          = "k8s-namespace"
+	metaKeyKubeServiceName = "k8s-service-name"
+)
+
+func GatewayToAPIGateway(k8sGW gwv1beta1.Gateway) consulAPI.APIGatewayConfigEntry {
+	listeners := make([]consulAPI.APIGatewayListener, 0, len(k8sGW.Spec.Listeners))
+	conditions := make([]consulAPI.Condition, 0, len(k8sGW.Status.Conditions))
+	for _, listener := range k8sGW.Spec.Listeners {
+		certificates := make([]consulAPI.ResourceReference, 0, len(listener.TLS.CertificateRefs))
+		for _, certificate := range listener.TLS.CertificateRefs {
+			c := consulAPI.ResourceReference{
+				Kind:        consulAPI.InlineCertificate,
+				Name:        string(certificate.Name),
+				SectionName: "",
+				Partition:   "",
+				Namespace:   string(*certificate.Namespace),
```

Comment:
and I'm assuming we'll need to do the same namespace mapping anywhere we use namespaces right?

Reply:
and if we use the consulIdentifier map we're just pushing up the namespace resolution and annotation overrides to the caller right?

Reply:
Yep, we'd be pushing it up to the caller, if we wanted I could even see doing something like generating a map of all of the names/namespaces/types we need created prior to the actual resource field mapping code just to make doing the name/namespace resolution easier. It'd require two passes across all of the resources we want generated, but I think it'll make some of the ref calculation code easier since we won't have to do any sort of annotation/namespace lookups there.

RE this:
> and I'm assuming we'll need to do the same namespace mapping anywhere we use namespaces right?

yep.